### PR TITLE
Fix ``geoips tree`` regression for workshop

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist

* [ ] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [ ] NO REQUIRED ***unit tests*** (Unit test is still not being ran for ``geiops tree``). I think we should chat about what we'd like to test here as it's largely hardcoded and probably not the best way to test this command.
* [ ] NO REQUIRED ***integration tests*** (explain why not required)
* [ ] Required ***documentation*** added for new/modified functionality
* [ ] Required ***release notes*** added for new/modified functionality
* [ ] NO REQUIRED ***updates to other repos*** (explain why not required)

# Related Issues
N/A. Needed for workshop right now.

# Testing Instructions
Run ``pytest -v /tests/unit_tests/commandline/`` and get all passing tests.

# Summary
With the addition of Command Class Factories, Shared Arguments, and Aliases, we had
an expected regression that would impact the functionality of the command
'geoips tree'. While we still may make updates to this command in the future,
regarding its output and how we'd like to unit test it, we now have working in a
much cleaner fashion. This PR addressed removing duplicate entries in the output
of the tree, as well hardcoded some conditionals for corner cases that would cause
duplicate entries do to the structure of aliases. For example, we had to add
conditionals when we encountered 'sector' as a command name, as this is an alias for
'geoips list sectors' and 'geoips describe sectors', and is the actual command name
of 'geoips test sector'.

Additionally, we commented out the functionality of 'hyperlinking' the text as the
current documentation is not being built up on GitHub and renders that functionality
useless at the moment. In the future, we expect to bring the hyperlink functionality
back in and update how we retrieve the command name as we change how 'prog' is set
for each commands' subparser attribute.

Files Modified:

    geoips/commandline/ancillary_info/alias_mapping.yaml
    geoips/commandline/geoips_describe.py
    geoips/commandline/geoips_list.py
    geoips/commandline/geoips_tree.py

# Output
```
(geoips) (base) [evan@spring geoips]$ geoips tree

geoips
    geoips config
        geoips config install
    geoips describe
        geoips describe algorithms
        geoips describe colormappers
        geoips describe coverage-checkers
        geoips describe feature-annotators
        geoips describe filename-formatters
        geoips describe gridline-annotators
        geoips describe interpolators
        geoips describe output-checkers
        geoips describe output-formatters
        geoips describe procflows
        geoips describe product-defaults
        geoips describe products
        geoips describe readers
        geoips describe sector-adjusters
        geoips describe sector-metadata-generators
        geoips describe sector-spec-generators
        geoips describe sectors
        geoips describe title-formatters
        geoips describe package
    geoips list
        geoips list algorithms
        geoips list colormappers
        geoips list coverage-checkers
        geoips list feature-annotators
        geoips list filename-formatters
        geoips list gridline-annotators
        geoips list interpolators
        geoips list output-checkers
        geoips list output-formatters
        geoips list procflows
        geoips list product-defaults
        geoips list products
        geoips list readers
        geoips list sector-adjusters
        geoips list sector-metadata-generators
        geoips list sector-spec-generators
        geoips list sectors
        geoips list title-formatters
        geoips list interfaces
        geoips list packages
        geoips list plugins
        geoips list scripts
        geoips list test-datasets
        geoips list unit-tests
    geoips run
        geoips run single_source
        geoips run data_fusion
        geoips run config_based
    geoips test
        geoips test linting
        geoips test script
        geoips test sector
    geoips tree
    geoips validate

WARNING: The GeoIPS CLI is currently under development and is subject to change.
Until this warning is removed, do not rely on the CLI to be static.
Please feel free to test the CLI and report any bugs or comments as an issue here:
https://github.com/NRLMMD-GEOIPS/geoips/issues/new/choose
```

### CODE REVIEW

File: geoips/commandline/geoips_tree.py [hunk 1]
```diff
@@ -144,15 +165,29 @@ def print_tree(self, parser, colored=False, level=0, max_depth=2, short_name=Fal
...
-        hyperlink = self.link(url, cmd_txt)
+        # Commenting these portions out until we can get the documentation build and
+        # make this easily testable. The introduction of Class Factories for commands
+        # has made this much harder to implement.
+
+        # url = f"{self.cmd_line_url}#{cmd_name.replace(' ', '-')}"
+        # hyperlink = self.link(url, cmd_txt)
+        hyperlink = cmd_txt
 
         if colored:
```
Comment (biosafetylvl5) [inline geoips/commandline/geoips_tree.py:192]: We should probably not use this word in code!

First dictionary entry: 

Colored is a racial descriptor historically used in the United States during the Jim Crow Era to refer to an African American. In many places, it may be considered a slur.

**Review by biosafetylvl5 — APPROVED** [review]
LGTM - but I didn't test it. Mindy - can you make sure to run full test on it?

File: geoips/commandline/geoips_tree.py (hunk 1 above)
Comment (evrose54) [reply geoips/commandline/geoips_tree.py:192]: Sure; I'll fixed that!

Comment (evrose54) [reply geoips/commandline/geoips_tree.py]: Fixed.

Comment (biosafetylvl5) [reply geoips/commandline/geoips_tree.py]: Thanks! :)

File: geoips/commandline/geoips_tree.py [hunk 2]
```diff
@@ -23,6 +23,27 @@ class GeoipsTree(GeoipsExecutableCommand):
...
+        So for 'list' (or 'ls'), there will be two 'geoips list' entries which we don't
+        want.
+        """
+        if not hasattr(self, "_cmd_aliases"):
+            self._cmd_aliases = []
+            for cmd_name in self.alias_mapping:
+                for alias in self.alias_mapping[cmd_name]:
+                    # Aliases below are names of actual commands and we need to deal
+                    # with this using conditionals in 'print_tree'
+                    if alias not in ["sector"]:
```
Comment (mindyls) [inline geoips/commandline/geoips_tree.py:43]: I'm not sure how I feel about the overlap between a command (sector) and an alias for an artifact (sectors interface -> sector alias.  Not sure if artifact is the right term here)

I think this is fine for now, but we may want to consider the potential ambiguity.

**Review by mindyls — APPROVED** [review]
Package: geoips_full
Total run time: 2036 seconds
Number data types run: 56
Number data types failed: 0
Wed Jul 31 19:14:53 UTC 2024

I tested this via my xarray dict branch (merged this in), full 0 return!

I left one comment, but this looks good to merge.

**Review by biosafetylvl5 — APPROVED** [review]

File: geoips/commandline/geoips_tree.py (hunk 2 above)
Comment (mindyls) [reply geoips/commandline/geoips_tree.py:43]: I'm resolving this

Comment (evrose54) [reply geoips/commandline/geoips_tree.py:43]: The main reason I see keeping ``sector`` as an alias is for the describe plugin command. For example, if we run ``geoips describe sector canada``, that makes more sense grammatically than ``geoips describe sectors canada``. Not sure how we should handle this one.
